spanner/client.go

```diff
@@ -635,6 +638,9 @@ func (c *Client) ReadOnlyTransaction() *ReadOnlyTransaction {
 	t.txReadOnly.disableRouteToLeader = true
 	t.txReadOnly.qo.DirectedReadOptions = c.dro
 	t.txReadOnly.ro.DirectedReadOptions = c.dro
+	if c.ro.LockHint == sppb.ReadRequest_LOCK_HINT_EXCLUSIVE {
+		t.txReadOnly.ro.LockHint = sppb.ReadRequest_LOCK_HINT_UNSPECIFIED
```

Comment:
This should be applied to `BatchReadOnlyTransaction` and `BatchReadOnlyTransactionFromID`

Reply:
Done